content/lectures/lec01.md

```diff
@@ -7,91 +7,193 @@ slides: https://docs.google.com/presentation/d/1FVklEogqEGn6zsp8YOpCuynUCCvhB6mX
+- Application: End user later (HTTP, FTP, SSH, DNS)
+- Presentation: Syntax layer (SSL, SSH, IMAP, FTP, JPEG)
+- Session: Synch and send to port (APIs, sockets)
+- Transport: End-to-end connections (TCP, UDP, QUIC)
+- Network: Packets (IP, ICMP, IPSec, IGMP)
+- Data Link: Frames (Ethernet, PPP, Switch, Bridge)
+- Physical: Physical structure (Coax, Fiber, Wireless, Hubs, Repeaters)
+
+## Internet Protocol (IP)
+
+IP is the lowest level of abstraction that we will uncover in this course. You can think of an IP address as a single atomic element that lives on a network. A computer is given a IP address that consists of four octets (32 bits of data) separated by dots. A subnet is a subset of all IP addresses available. The IP ranges given for Loopback and Private IPs below represent subnets, but a subnet can be any range of IPs.
+
+There are a few special IP addresses that you might become familiar with:
+- Loopback: IP addresses on the local machine.
+  - 127.0.0.0 - 127.255.255.255 (127.0.0.0/8)
+- Private: Only for devices on the inside of the local network, are never surfaced publicly
+  - 10.0.0.0 - 10.255.255.255 (10.0.0.0/8)
+  - 172.16.0.0 - 172.31.255.255 (172.16.0.0/12)
+  - 192.168.0.0 - 192.168.255.255 (192.168.0.0/16)
+  
+If you'd like to see your IP address on your machine, you can run one of these commands depending on your OS:
+
+- Linux: `ip addr show`
+- Mac: `ifconfig`
+- Windows: `ipconfig \all`
+
+
+## Visualizing the Network
+
+Say we have two machines on a subnet, this means that these machines know each other exist, and they can communicate, but they don't know _how_ to communicate. They need some shared language that will allow them to send information back and forth between them.
+
+## A Note on IPv6
+
+The four octets in IPv4 only gives us 32 bits of data to specify an IP address, this gives us 2^(32) = 4,294,967,296 possible IP addresses. When IPv4 was invented that seemed like more than we would ever need, but nowadays every smartphone, TV, PlayStation, or laptop might need an IP address. The solution was IPv6 which allowed for 128 bits of data in IP addresses, drastically expanding the number of available IP addresses.
+
+However, IPv6 has really struggled with adoption because network engineers aren't too keen on migrating all of their technology to IPv6. 45% of network traffic in the US happens over IPv6, but that's because many big tech companies have adopted IPv6 on their internal networks. Outside of that, few people have switched over, so we'll use IPv4 throughout this class.
+
+
+## Transport Layer: TCP
+
+TCP is this shared language that is layered on top of IP to allow for a shared language between computers. TCP has robust error-checking to ensure that all of the information transmitted over the internet actually gets to the destination. This error-checking can also cause a delay though as it requires the person receiving the data to also send back a confirmation that they got everything.
```

Comment:
I don't know if delay is the right word here. Maybe reference it adding overhead or reducing bandwidth? @pawalt suggestions?

Reply:
Delay is more correct than bandwidth since it implies latency issues, not throughput issues.
The most correct phrasing is probably "cause latency and congestion issues"